src/Spectre.Data/RoiIo/RoiReader.cs

```diff
@@ -0,0 +1,107 @@
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+namespace Spectre.Data.RoiIo
+{
+    using System.Collections.Generic;
+    using System.Drawing;
+    using System.IO;
+    using System.Linq;
+    using Datasets;
+
+    /// <summary>
+    /// Provides listing all available ROI's from directory.
+    /// Provides reading a ROI from a directory.
+    /// </summary>
+    public class RoiReader
+    {
+        /// <summary>
+        /// The path to the Rois directory.
+        /// </summary>
+        private string _path;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RoiReader"/> class.
+        /// </summary>
+        /// <param name="path">The path.</param>
+        public RoiReader(string path)
+        {
+            _path = path;
+        }
+
+        /// <summary>
+        /// Lists the rois from directory.
+        /// </summary>
+        /// <returns>
+        /// All ROIs in specified directory.
+        /// </returns>
+        /// <exception cref="FileNotFoundException">No *.png files found in directory or subdirectories</exception>
+        public List<Roi> GetAllRoisFromDirectory()
+        {
+            var rois = new List<Roi>();
+
+            var names = GetAllNamesFromDirectory();
+
+            for (var listIterator = 0; listIterator < names.Count; listIterator++)
+            {
+                rois.Add(GetSingleRoiFromDirectory(Path.Combine(_path, names[listIterator])));
+            }
+
+            return rois;
+        }
+
+        /// <summary>
+        /// Gets names of all ROIs from directory.
+        /// </summary>
+        /// <returns>
+        /// Returns names of all files in specified directory.
+        /// </returns>
+        /// <exception cref="FileNotFoundException">No *.png files found in directory.</exception>
+        public List<string> GetAllNamesFromDirectory()
+        {
+            var names = new List<string>();
+
+            var allfiles = Directory.GetFiles(_path, "*.png", SearchOption.TopDirectoryOnly);
+
+            names = allfiles.ToList();
+            names.Sort();
+
+            return names;
+        }
+
+        /// <summary>
+        /// Loads single ROI from specified folded.
+        /// </summary>
+        /// <param name="fileName">Name of the file with extension.</param>
+        /// <returns>
+        /// ROI dataset.
+        /// </returns>
+        public Roi GetSingleRoiFromDirectory(string fileName)
+        {
+            var roiConverter = new RoiConverter();
```

Comment:
We're recreating this for each `Roi` read. Can we create it once, at creation of `RoiReader`?